src/policy/CMSRucioPolicy/permission.py

```diff
@@ -167,6 +174,128 @@ def perm_update_rse(issuer, kwargs, *, session: "Optional[Session]" = None):
+        return False
+
+    account = kwargs['account']
+    auto_approve_activity = 'User AutoApprove'
+    dids = kwargs['dids']
+    try:
+        global_usage_all_accounts = float(config_get(
+            'rules', 'global_usage_all_accounts', raise_exception=True, default=1e16))
+    except (NoOptionError, NoSectionError, RuntimeError):
+        global_usage_all_accounts = 1e16
+
+    try:
+        global_usage_per_account = float(config_get(
+            'rules', 'global_usage_per_account', raise_exception=True, default=1e15))
+    except (NoOptionError, NoSectionError, RuntimeError):
+        global_usage_per_account = 1e15
+
+    try:
+        rule_lifetime_threshold = int(config_get('rules', 'rule_lifetime_threshold',
+                                      raise_exception=True, default=2592000))
+    except (NoOptionError, NoSectionError, RuntimeError):
+        rule_lifetime_threshold = 2592000
+
+    try:
+        single_rse_rule_size_threshold = float(config_get(
+            'rules', 'single_rse_rule_size_threshold', raise_exception=True, default=50e12))
+    except (NoOptionError, NoSectionError, RuntimeError):
+        single_rse_rule_size_threshold = 50e12
+
+    # Check if the account is banned
+    if has_account_attribute(account, 'rule_banned', session=session):
+        return False
+
+    # Check if the rule is locked
+    if kwargs['locked']:
+        return False
+
+    if kwargs['lifetime'] is None:
+        return False
+
+    if kwargs['lifetime'] > rule_lifetime_threshold:
+        return False
+
+    for did in dids:
+        size_of_rule = sum([file['bytes']
+                            for file in list_files(InternalScope(did['scope']),
+                                                   did['name'],
+                                                   session=session)])
+
+        # Limit single RSE rules to 50 TB
+        # This does not mean that the total locks size at a RSE will be limited to 50 TB
+        # as other rules that are spread over multiple RSEs may claim the same space
+        # This is just a simple check to avoid a single RSE rules from being too large
+        rse_expression = kwargs['rse_expression']
+        rses = parse_expression(rse_expression, filter_={'availability_write': True}, session=session)
+
+        if len(rses) == 1:
+            this_rse_autoapprove_rules = list_rules(
+                filters={'account': account, 'activity': auto_approve_activity, 'rse_expression': rse_expression},
+                session=session)
+            this_rse_autoapprove_usage = _get_rule_size(this_rse_autoapprove_rules)
+            if this_rse_autoapprove_usage + size_of_rule > single_rse_rule_size_threshold:
+                logging.warning(
+                    'Single RSE usage exceeded for auto approve rules for account %s and RSE %s, this_rse_autoapprove_usage, size_of_rule, single_rse_rule_size_threshold: %s, %s, %s',
```

Comment:
use an f-string here. Much clearer

Reply:
I use `%s` formatting in log functions to keep pylint happy. https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html
If you think I should change to fstring for readability, let me know. I will change it.

Reply:
Oh, yeah. This is a performance issue more than anything. Up to you I guess. The f-string way will give a warning and possibly perform more poorly since you have to build strings for messages not emitted. But in this case, that's not an issue since all warning messages are going to be output.

Reply:
Ok, then I would let it be.